src/frontend/src/components/FillForms/UserDataForm.tsx

```diff
@@ -1,337 +1,293 @@
-        return nameString;
-      } else {
-        return "";
-      }
-    } else {
-      return "";
-    }
-  };
+const isPhoneNumber = (str: string) => {
+  //the phone RegEx accepts the following formats (123) 456-7890, 123-456-7890, 123.456.7890, 1234567890, (123)-456-7890
+  return /^(\(?\d{3}\)?[-.\s]?){2}\d{4}$/.test(str); // Check for empty string included
+};
 
-  private buildDob = () => {
-    if (this.props.aliases.length > 0) {
-      const firstAlias = this.props.aliases[0];
-      return firstAlias.birth_date;
-    } else {
-      return "";
-    }
-  };
+const isZipCode = (str: string) => {
+  //the zipCode RegEx accepts any 5 digit entry ex. "12345"
+  return /[0-9][0-9][0-9][0-9][0-9]/.test(str); // Check for empty string included
+};
 
-  public state: State = {
-    name: this.buildName(),
-    dob: this.buildDob(),
-    mailingAddress: "",
-    phoneNumber: "",
-    city: "",
-    state: "",
-    zipCode: "",
-    invalidZipCode: false,
-    invalidPhone: false,
-    invalidBirthDate: false,
-  };
+export default function UserDataForm() {
+  const aliases = useAppSelector(selectSearchFormValues).aliases;
+  const dispatch = useDispatch();
+  const [name, setName] = useState(buildName());
+  const [dob, setDob] = useState(aliases[0].birth_date);
+  const [mailingAddress, setMailingAddress] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [city, setCity] = useState("");
+  const [state, setState] = useState("Oregon");
+  const [zipCode, setZipCode] = useState("");
+  const [errorMessages, setErrorMessages] = useState<string[]>([]);
+  const loadingExpungementPacket = useAppSelector(
+    (state) => state.search.loadingExpungementPacket
+  );
 
-  phoneNumberPattern = new RegExp(".*[0-9].*");
-  zipCodePattern = new RegExp("[0-9][0-9][0-9][0-9][0-9].*");
+  const [showModal, setShowModal] = useState(false);
 
-  public validateForm = () => {
-    const phoneNumberMatch = this.phoneNumberPattern.exec(
-      this.state.phoneNumber
-    );
-    const zipCodeMatch = this.zipCodePattern.exec(this.state.zipCode);
-    return new Promise((resolve: (value?: unknown) => void) => {
-      this.setState<any>(
-        {
-          invalidZipCode: this.state.zipCode.length > 0 && !zipCodeMatch,
-          invalidPhone: this.state.phoneNumber.length > 0 && !phoneNumberMatch,
-          invalidBirthDate:
-            this.state.dob.length > 0 &&
-            moment(this.state.dob, "M/D/YYYY", true).isValid() === false,
-        },
+  function buildName() {
+    const { first_name, middle_name, last_name } = aliases[0];
+    let fullName = "";
+    fullName += first_name ? first_name + " " : "";
+    fullName += middle_name ? middle_name + " " : "";
+    fullName += last_name;
+
+    return fullName.includes("*") ? "" : fullName || " ";
+  }
+
+  function allInputsValid() {
+    const errorMessages = {
+      "Date Birth format must be mm/dd/yyyy": isPresent(dob) && !isDate(dob),
```

Comment:
Is this format something that the courts have asked for? I could see someone wanting to just type out "Jan 1, 2023" - after all we aren't parsing this value but writing this out onto a form that would be read by a human.

Reply:
I would love to demo but I'm not sure what a silent screen demo is. Is it screenshots or video? 

As far as the date format, that's how it was before I touched it and I'm not sure if that's the required format for the courts. Anytime I've ever dealt with dates I've always used something like <input type="date"> to force the user into a uniform format...

I'm happy to demo and/or change anything 😊

Reply:
> Is it screenshots or video?
video

> that's how it was before I touched it and I'm not sure if that's the required format for the courts
Let's leave it out then. It gets auto-filled from the previous screen and the previous screen requires that format but that isn't for the courts.

Reply:
alright I made the vid. I'm not sure how to attach it here so I put it on my google drive and I'll link it [here](https://drive.google.com/file/d/1t-OQYYK8NI__k8AJq-LATWypODducfK4/view?usp=sharing). I can always share it on discord/slack if that's better.

P.S. I noticed that the error messages were lingering even after a successful submit as well as the error message for the phone number being a bit odd. I'm going to push those two small changes now